galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager/src/main/java/dev/galasa/zos3270/spi/Zos3270TerminalImpl.java

```diff
@@ -205,45 +217,155 @@ public synchronized void screenUpdated(Direction direction, AttentionIdentificat
-            // sort
-            // out
-            // alt
-            // sizes
-            dev.galasa.zos3270.common.screens.Terminal rasTerminal = new dev.galasa.zos3270.common.screens.Terminal(
-                    this.terminalId, this.runId, rasTerminalSequence, terminalSize);
-            rasTerminal.getImages().addAll(this.cachedImages);
-
-            JsonObject intermediateJson = (JsonObject) gson.toJsonTree(rasTerminal);
-            stripFalseBooleans(intermediateJson);
-            String tempJson = gson.toJson(intermediateJson);
-
-            if (applyCtf) {
-                tempJson = cts.removeConfidentialText(tempJson);
-            }
-
-            String terminalFilename = this.terminalId + "-" + String.format("%05d", rasTerminalSequence) + ".gz";
-            Path terminalPath = terminalRasDirectory.resolve(terminalFilename);
-
-            try (GZIPOutputStream gos = new GZIPOutputStream(Files.newOutputStream(terminalPath,
-                    new SetContentType(new ResultArchiveStoreContentType("application/zos3270terminal")),
-                    StandardOpenOption.CREATE))) {
-                IOUtils.write(tempJson, gos, "utf-8");
+            writeTerminalGzJson();
+            if (writeImages) {
+                writeTerminalImage();
             }
         } catch (Exception e) {
             logger.error("Unable to write terminal cache to the RAS", e);
             rasTerminalSequence--;
             return;
         }
+    }
+
+    /**
+     * This method creates png images to represent the Terminal screens and writes them to the RAS
+     * @throws IOException
+     */
+    private synchronized void writeTerminalImage() throws IOException {
+        TerminalSize terminalSize = new TerminalSize(getScreen().getNoOfColumns(), getScreen().getNoOfRows());
+        dev.galasa.zos3270.common.screens.Terminal rasTerminal = new dev.galasa.zos3270.common.screens.Terminal(
+                this.terminalId, this.runId, rasTerminalSequence, terminalSize);
+        rasTerminal.getImages().addAll(this.cachedImages);
+
+        // 2 extra rows added for Inbound/Outbound info and extra space
+        int numRows = terminalSize.getRows() + 2;
+        int numCols = terminalSize.getColumns();
+
+        int width = numCols * 7;
+        int height = numRows * 13; 
+
+        BufferedImage image = new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
+        Graphics2D graphics = image.createGraphics();
+
+        Font font = new Font(Font.MONOSPACED, Font.PLAIN, 10);
+        graphics.setFont(font);
+        FontMetrics fontMetrics = graphics.getFontMetrics();
+        int fontHeight = fontMetrics.getHeight();
+        int fontWidth = fontMetrics.getMaxAdvance();
+
+        graphics.setPaint(Color.black);
+        graphics.fillRect(0, 0, width, height);
+        graphics.setPaint(Color.green);
+
+        List<TerminalImage> terminalImages = rasTerminal.getImages();
```

Comment:
why do we get this from the rasTerminal?  Couldn't we reference the cached images directly?